docs/data/data-grid/row-ordering/RowReorderingValidation.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import {
+  DataGridPro,
+  GridValidRowModel,
+  ReorderValidationContext,
+} from '@mui/x-data-grid-pro';
+import { useDemoData } from '@mui/x-data-grid-generator';
+
+const isValidRowReorder = (context: ReorderValidationContext) => {
+  if (context.sourceNode.type === 'group') {
+    return true;
+  }
+  if (
+    context.targetNode.type === 'group' &&
+    context.targetNode.id !== context.sourceNode.parent
```

Comment:
```suggestion
    context.targetNode.id !== context.sourceNode.parent &&
    context.dropPosition === 'inside'
```

Otherwise, it allows to reorder leaves inside of the group, which doesn't reflect the intention here if I understand correctly:

> The demo below prohibits reordering leaf rows under the same parent and only allows cross-parent operations